### PR TITLE
Add collation and config check

### PR DESCRIPTION
**Pull request by dimaursu**
This should take care of #5616 and a bunch of other complaints from our podmins.

### CODE REVIEW

File: db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb [hunk 1]
```diff
@@ -5,23 +5,32 @@ class SetMysqlToUnicodeMb4 < ActiveRecord::Migration
...
-    change_encoding('utf8') if AppConfig.mysql?
+    change_encoding('utf8', 'utf8_bin') if AppConfig.mysql?
   end
 
-  def change_encoding(encoding)
-    execute "ALTER DATABASE `#{ActiveRecord::Base.connection.current_database}` CHARACTER SET #{encoding};"
+  def check_config(encoding, collation)
+    connection_config = ActiveRecord::Base.connection_config
+    fail "Database encoding is not #{encoding}!"   if connection_config[:encoding] != encoding
+    fail "Database collation is not #{collation}!" if connection_config[:collation] != collation
```
Comment (jhass) [inline db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb]: `fail` is used nowhere in our codebase, please use `raise`.

Comment (dimaursu) [inline db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb]: we should start recommend using it :-)

Comment (jhass) [inline db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb]: But keep consistency for now.

**Comment by rivendale2010** [comment]
randomtemp

Yay! :)

**Comment by jhass** [comment]
Let's try it.

**Pull request merged**